hazelcast/src/main/java/com/hazelcast/query/impl/CachedQueryEntry.java

```diff
@@ -102,7 +102,10 @@ protected Object getTargetObject(boolean key) {
             }
         } else {
             if (valueObject == null) {
-                if (valueData.isPortable()) {
+                // valueObject and valueData are null when the default constructor is used and init() is not yet called
+                if (valueData == null) {
+                    return null;
+                } else if (valueData.isPortable()) {
```

Comment:
This is the line which produces a NPE if you run `getTargetObject()` on an `CachedQueryEntry` which was not initialized.